homeassistant/components/switchbot_via_api/config_flow.py

```diff
@@ -0,0 +1,75 @@
+"""Config flow for Switchbot via API integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from switchbot_api import SwitchBotAPI
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_API_KEY, CONF_API_TOKEN
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.exceptions import HomeAssistantError
+
+from .const import DOMAIN, ENTRY_TITLE
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_API_TOKEN): str,
+        vol.Required(CONF_API_KEY): str,
+    }
+)
+
+
+async def validate_input(hass: HomeAssistant, data: dict[str, Any]) -> dict[str, Any]:
+    """Validate the user input allows us to connect.
+
+    Data has the keys from STEP_USER_DATA_SCHEMA with values provided by the user.
+    """
+    await SwitchBotAPI(
+        token=data[CONF_API_TOKEN], secret=data[CONF_API_KEY]
+    ).list_devices()
+    return data
+
+
+@config_entries.HANDLERS.register(DOMAIN)
+class SwitchbotViaAPIConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Switchbot via API."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+        errors: dict[str, str] = {}
+        if user_input is not None:
+            try:
+                info = await validate_input(self.hass, user_input)
+            except CannotConnect:
+                errors["base"] = "cannot_connect"
+            except InvalidAuth:
+                errors["base"] = "invalid_auth"
+            except Exception:  # pylint: disable=broad-except
+                _LOGGER.exception("Unexpected exception")
+                errors["base"] = "unknown"
+            else:
+                await self.async_set_unique_id(DOMAIN, raise_on_progress=False)
```

Comment:
https://developers.home-assistant.io/docs/config_entries_config_flow_handler?_highlight=unique&_highlight=id&_highlight=req#unique-id-requirements

This doesn't meet the unique id requirements

Reply:
Ok, I will use the token hashed then as it is not changed when resetting the token in the app.